### ISSUE TITLE
Issues with 0.37

### ISSUE THREAD

**Issue by QuentinFchx** [bbatsov/rubocop#2773]
```
An error occurred while Style/SpaceAroundOperators cop was inspecting (...)
undefined method `source_range' for nil:NilClass
lib/rubocop/cop/style/space_around_operators.rb:46:in `on_send'
```

```
An error occurred while Style/MultilineOperationIndentation cop was inspecting (...)
undefined method `dot' for #<Parser::Source::Map::Keyword:0x007fcb6b715f00>
lib/rubocop/cop/style/dot_position.rb:12:in `on_send'
```

```
An error occurred while Lint/DefEndAlignment cop was inspecting (...)
undefined method `size' for nil:NilClass
lib/rubocop/cop/mixin/on_method_def.rb:26:in `modifier_and_def_on_same_line?'
```

```
An error occurred while Lint/UselessComparison cop was inspecting (...)
undefined method `selector' for #<Parser::Source::Map::Keyword:0x007fcb6b715f00>
lib/rubocop/cop/lint/useless_comparison.rb:19:in `on_send'
```

And 3-4 others of the same kind

**Comment by bbatsov** [comment]
Without actual code triggering those errors we can't really help you.

**Comment by QuentinFchx** [comment]
I thought it could be a more general issue since it impacts different files and cops.
I'll try to find what is causing the issue (can't post it here since it's core business related).

**Comment by QuentinFchx** [comment]
https://gist.github.com/QuentinFchx/e76e9a3617c16c395a05
Here is a file triggering 9 errors, along with our rubocop configuration

**Comment by mikestok** [comment]
```
class Foo < Bar
  def method(arg)
    super(arg) do
      something
    end
  end
end
```

triggers

```
Offenses:

bug.rb:1:1: C: Missing top-level class documentation comment.
class Foo < Bar
^^^^^
bug.rb:3:10: C: Do not use parentheses for method calls with no arguments.
    super(arg) do
         ^

1 file inspected, 2 offenses detected

8 errors occurred:
An error occurred while Lint/DefEndAlignment cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Lint/UselessComparison cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Style/ColonMethodCall cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Style/DotPosition cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Style/IndentationWidth cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Style/MultilineMethodCallIndentation cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Style/MultilineOperationIndentation cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
An error occurred while Style/SpaceAroundOperators cop was inspecting /Users/mike/tmp/rubobug/bug.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.37.0 (using Parser 2.3.0.2, running on ruby 2.3.0 x86_64-darwin15)
```

**Comment by alexdowad** [comment]
Please run with the `-d` flag to get a stack trace for each error.

**Comment by rilian** [comment]
with cop `Performance/RedundantMerge` I get a recommendation to replace

```ruby
Invoice.all.each { |i| i.update(customer_id: i.job.customer_id) }
```
to 
```ruby
Invoice.all.each { |i| i[:customer_id] = i.job.customer_id }
```
which is wrong, since `update` saves an object, and simple assignment - not

**Comment by basex** [comment]
I have the same problem. I have a project with ActiveRecord and DataMapper and I got automated corrections from ```update(customer_id: i.job.customer_id)``` to ```i[:customer_id] = i.job.customer_id```. When using an ORM this is clearly different.

**Comment by mikestok** [comment]
```
# a comment
class Foo < Bar
  def method(arg)
    super(arg) do
      something
    end
  end
end
```

yields this https://gist.github.com/e424f18450eba7336e80

**Comment by maxjacobson** [comment]
@alexdowad here are some stack traces from 0.37:

```
bin/rubocop --only IndentationWidth -d app/controllers/sessions_controller.rb 
For /Users/max/src/evolve: configuration from /Users/max/src/evolve/.rubocop.yml
Inheriting configuration from /Users/max/src/evolve/.rubocop_todo.yml
Default configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/default.yml
Inheriting configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/enabled.yml
Inheriting configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/disabled.yml
Inspecting 1 file
Scanning /Users/max/src/evolve/app/controllers/sessions_controller.rb
An error occurred while Style/IndentationWidth cop was inspecting /Users/max/src/evolve/app/controllers/sessions_controller.rb.
undefined method `size' for nil:NilClass
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/mixin/on_method_def.rb:26:in `modifier_and_def_on_same_line?'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/style/indentation_width.rb:86:in `on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:39:in `block (2 levels) in on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:97:in `with_cop_error_handling'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:38:in `block in on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:37:in `each'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:37:in `on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:152:in `on_block'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_block'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:96:in `on_def'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_def'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:89:in `on_class'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_class'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:13:in `walk'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:56:in `investigate'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/team.rb:58:in `inspect_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:194:in `inspect_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:164:in `block in do_inspection_loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:158:in `loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:158:in `do_inspection_loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:87:in `process_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:59:in `block in inspect_files'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:57:in `each'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:57:in `inspect_files'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:35:in `run'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cli.rb:30:in `run'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/bin/rubocop:14:in `block in <top (required)>'
/Users/max/.rubies/2.3.0/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/bin/rubocop:13:in `<top (required)>'
bin/rubocop:16:in `load'
bin/rubocop:16:in `<main>'
```

```
bin/rubocop --only DefEndAlignment -d app/models/manual_interaction_attempt.rb
For /Users/max/src/evolve: configuration from /Users/max/src/evolve/.rubocop.yml
Inheriting configuration from /Users/max/src/evolve/.rubocop_todo.yml
Default configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/default.yml
Inheriting configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/enabled.yml
Inheriting configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/disabled.yml
Inspecting 1 file
Scanning /Users/max/src/evolve/app/models/manual_interaction_attempt.rb
An error occurred while Lint/DefEndAlignment cop was inspecting /Users/max/src/evolve/app/models/manual_interaction_attempt.rb.
undefined method `size' for nil:NilClass
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/mixin/on_method_def.rb:26:in `modifier_and_def_on_same_line?'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/lint/def_end_alignment.rb:31:in `on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:39:in `block (2 levels) in on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:97:in `with_cop_error_handling'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:38:in `block in on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:37:in `each'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:37:in `on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:152:in `on_block'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_block'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:96:in `on_def'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_def'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:89:in `on_class'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_class'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:13:in `walk'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:56:in `investigate'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/team.rb:58:in `inspect_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:194:in `inspect_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:164:in `block in do_inspection_loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:158:in `loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:158:in `do_inspection_loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:87:in `process_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:59:in `block in inspect_files'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:57:in `each'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:57:in `inspect_files'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:35:in `run'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cli.rb:30:in `run'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/bin/rubocop:14:in `block in <top (required)>'
/Users/max/.rubies/2.3.0/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/bin/rubocop:13:in `<top (required)>'
bin/rubocop:16:in `load'
bin/rubocop:16:in `<main>'
```

And:

```
bin/rubocop --only UselessComparison -d app/controllers/sessions_controller.rb
For /Users/max/src/evolve: configuration from /Users/max/src/evolve/.rubocop.yml
Inheriting configuration from /Users/max/src/evolve/.rubocop_todo.yml
Default configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/default.yml
Inheriting configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/enabled.yml
Inheriting configuration from /Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/config/disabled.yml
Inspecting 1 file
Scanning /Users/max/src/evolve/app/controllers/sessions_controller.rb
An error occurred while Lint/UselessComparison cop was inspecting /Users/max/src/evolve/app/controllers/sessions_controller.rb.
undefined method `selector' for #<Parser::Source::Map::Keyword:0x007fd543b23a70>
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/lint/useless_comparison.rb:19:in `on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:39:in `block (2 levels) in on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:97:in `with_cop_error_handling'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:38:in `block in on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:37:in `each'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:37:in `on_send'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:152:in `on_block'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_block'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:96:in `on_def'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_def'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_begin'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:89:in `on_class'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:43:in `on_class'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/ast_node/traversal.rb:13:in `walk'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/commissioner.rb:56:in `investigate'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cop/team.rb:58:in `inspect_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:194:in `inspect_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:164:in `block in do_inspection_loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:158:in `loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:158:in `do_inspection_loop'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:87:in `process_file'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:59:in `block in inspect_files'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:57:in `each'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:57:in `inspect_files'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/runner.rb:35:in `run'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/lib/rubocop/cli.rb:30:in `run'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/bin/rubocop:14:in `block in <top (required)>'
/Users/max/.rubies/2.3.0/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
/Users/max/.gem/ruby/2.3.0/gems/rubocop-0.37.0/bin/rubocop:13:in `<top (required)>'
bin/rubocop:16:in `load'
bin/rubocop:16:in `<main>'
```

**Comment by alexdowad** [comment]
Dear users, in future, please open a separate issue for each bug, rather than combining many bug reports in a single issue.

**Comment by alexdowad** [comment]
@mikestok, thanks for the sample code. All the bugs there are fixed in my open PR.

**Comment by alexdowad** [comment]
@QuentinFchx, all your bugs are also fixed in my open PR. @bbatsov @jonas054, please close this issue.

**Comment by QuentinFchx** [comment]
Awesome, thanks @alexdowad !

**Comment by maxjacobson** [comment]
Everything is awesome now, thanks :smile:

**Comment by mikestok** [comment]
Thank you all!

> On Feb 9, 2016, at 11:49 AM, Bozhidar Batsov <notifications@github.com> wrote:
> 
> Closed #2773 <https://github.com/bbatsov/rubocop/issues/2773>.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/bbatsov/rubocop/issues/2773#event-544603128>.
> 

-- 

Mike Stok <mike@stok.ca>
http://www.stok.ca/~mike/

The "`Stok' disclaimers" apply.

**Comment by basex** [comment]
Working great. Thanks :)
